docs/CaliptraIntegrationSpecification.md

```diff
@@ -725,6 +725,145 @@ The following code snippet and schematic diagram illustrate JTAG originating CDC
+## Reset Sequencing Scenarios
+
+The resets defined in *Table 20* has the following sequencing phases which are applicable for different reset scenarios - cold boot, cold reset, warm reset and firmware reset. 
+
+The reset sequencing looks like below.
+
+![](./images/reset_sequencing.png)
+
+## Reset Ordering
+
+The table below defines the order in which reset can get asserted. A ">>" in a cell at row X and column Y indicates that if the reset in row X is asserted, the reset is row Y will also get asserted. For rest of the cells (in which symbol ">>" is not there) the above assumption doesn't hold good and hence the paths between those resets are potential RDC violations. The black cells can be ignored as they are between same resets.
+
+![](./images/reset_ordering.png)
+
+## Reset Constraints and Assumptions
+
+Below are the set of constraints and assumptions that will need to be provided before running RDC structural analysis of Caliptra Core IP.
+
+1. Caliptra Core IP assumes the primary reset inputs are already synchronized to their respective clocks. The integrator should add external reset synchronizers to achieve the same.
+    - *cptra_pwrgood* and *cptra_rst_b* resets should be synchronzied to *cptra_clk*
+    - *jtag_trst_n* reset should be synchronzied to *jtag_clk*
+2. The following debug registers which are driven from JTAG will not be toggled during functional flow.
+    - u_caliptra.rvtop.veer.dbg.dmcontrol_reg[0] = 0
+3. Set *scan_mode* to 0 for functional analysis.
+4. Stamp/Create functional resets for *cptra_noncore_rst_b* and *cptra_uc_rst_b* at the definition points as mentioned in *Table 20*. 
+5. Create funtional reset grouping - This step need to be customized as per the EDA tool which is being used for RDC analysis. Basically the thought process here is to achieve the below three sequencing requirements/constraints:-
+    - Gate all clocks when *cptra_noncore_rst_b* is asserted. This will ensure that the capture flop clock is gated while the source flop's reset is getting asserted, therby preventing it from going metastable. So when *cptra_noncore_rst_b* is going to get asserted, below signals are constrained to be at 1 around that time.
+        - soc_ifc_top1.i_soc_ifc_boot_fsm.rdc_clk_dis
+        - soc_ifc_top1.i_soc_ifc_boot_fsm.arc_IDLE
+    - Gate Veer core clock when *cptra_uc_rst_b* is asserted. This will ensure that the capture flop clock is gated while the source flop's reset is getting asserted, therby preventing it from going metastable. So when *cptra_uc_rst_b* is going to get asserted, below signal is constrained to be at 1 around that time.
+        - soc_ifc_top1.i_soc_ifc_boot_fsm.fw_update_rst_window
+    - Quiesce AHB bus before *cptra_uc_rst_b* is asserted. Since firmware reset request is triggered by SW, it can ensure that all AHB transcations are flushed out before initiating a reset request. So when *cptra_uc_rst_b* is going to get asserted, below signal is constrained to be at 0 around that time.
```

Comment:
Could we add description of the methodology for forcing AHB bus to quiesce on occurrence of generated resets? I.e., use of reset windows as in https://github.com/chipsalliance/caliptra-rtl/blob/839800557abf621909d95e752553a52cd0957008/src/integration/rtl/caliptra_top.sv#L328

Reply:
added details about the process of generating window, please check